app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.mission
+
+import ch.epfl.sdp.drone3d.drone.DroneService
+import com.mapbox.mapboxsdk.geometry.LatLng
+import java.lang.IllegalArgumentException
+import javax.inject.Inject
+
+class ParallelogramMappingMissionService @Inject constructor(override val droneService: DroneService):MappingMissionService{
+
+    private val cameraAngle = 0.0 // Suppose the drone is looking down
+
+    override fun buildSinglePassMappingMission(vertices:List<LatLng>,flightHeight:Double): List<LatLng>? {
+
+        return buildMappingMission(
+            vertices,
+            flightHeight,
+            ParallelogramMissionBuilder.Companion::buildSinglePassMappingMission
+        )
+    }
+
+    override fun buildDoublePassMappingMission(vertices:List<LatLng>,flightHeight:Double): List<LatLng>? {
+        return buildMappingMission(
+            vertices,
+            flightHeight,
+            ParallelogramMissionBuilder.Companion::buildDoublePassMappingMission
+        )
+    }
+
+    private fun buildMappingMission(
+        vertices: List<LatLng>,
+        flightHeight: Double,
+        mappingFunction: (
+            startingPoint: Point, area: Parallelogram, cameraAngle: Double,
+            flightHeight: Double, groundImageDimension: GroundImageDim
+        ) -> List<Point>
+    ): List<LatLng>? {
+
+        assert(vertices.size==3)
```

Comment:
Assert should not be used in android. Prefere if (...) thrown IllegalArguementException()

Reply:
Corrected